test/unit/search/search_result_range_test.cpp

```diff
@@ -21,16 +21,42 @@
 // Simple executor used as mock for the test.
 // ----------------------------------------------------------------------------
 
+struct dummy_result_type
+{
+    size_t query_id_{};
+    size_t reference_begin_pos_{};
+
+    friend bool operator==(dummy_result_type const & lhs, dummy_result_type const & rhs)
+    {
+        return std::tie(lhs.query_id_, lhs.reference_begin_pos_) == std::tie(rhs.query_id_, rhs.reference_begin_pos_);
+    }
+
+    friend bool operator!=(dummy_result_type const & lhs, dummy_result_type const & rhs)
+    {
+        return !(lhs == rhs);
+    }
+
+    friend std::ostream & operator<<(std::ostream & os, const dummy_result_type & res)
```

Comment:
why is this needed?

Reply:
Namings

Reply:
Because the expect_ranges_eq requires the type it uses to be pretty_printable